### PR TITLE
Update Proxy-Support check to be case insensitive

### LINKED ISSUE #61414: Proxy-Support header case sensitive causes proxied API failure when using HttpClient
I've been looking into an issue where access to a API work via PowerShell but doesn't via HttpClient.

The diagnostic output logs "Proxy doesn't support connection-based auth" and looking it to that appear to be proxy requires the header "Proxy-Support" with the value "Session-Based-Authentication".
https://github.com/dotnet/runtime/blob/e64cce6fbbde72d1ea61010bd08f11ba7b51afc9/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs#L64

However the API in question has the "Proxy-Support" with the value "Session-based-authentication" a simple case change.  While I can accept the specification for the proxy support header may specify the value with an exact case, following this means dotnet apps will not work with some API, and other implementations do not enforce the casing.

Suggested fix would be to make the check case insensitive.

### PR DESCRIPTION
**Pull request by ChrisFWood**
RFC4559 does not specify that the Proxy-Support header value used to
determine if the proxy server will honour client server authentication
integrity is case sensitive. Updating the check to be case insensitive
to prevent failures when value is supplied using differences in case.

Fix #61414

### CODE REVIEW

**Review by karelz — APPROVED** [review]
Thanks for the fix @ChrisFWood

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs [hunk 1]
```diff
@@ -61,7 +61,7 @@ private static bool ProxySupportsConnectionAuth(HttpResponseMessage response)
 
             foreach (string v in values)
             {
-                if (v == "Session-Based-Authentication")
```
Comment (stephentoub) [inline src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs:64]: Separately, @geoffkizer, we might want to audit use of Headers and update appropriate sites to use NonValidated.

**Review by stephentoub — APPROVED** [review]

**Comment by danmoseley** [comment]
Is a test feasible?

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs (hunk 1 above)
Comment (geoffkizer) [reply src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs:64]: Is the goal here to improve performance by avoiding the added processing/memory use that happens when you go through TryGetValues etc?

There are semantic differences with NonValidated. In particular if the server sent something like 

```
Proxy-Support: Session-Based-Authentication, Some-Other-Magic-Token
```

Then the current code would enumerate this as two header values and successfully match the first. In general for header processing, this is what you want (e.g. consider Connection: close handling).

If we just use NonValidated here then we would need to parse these tokens ourselves. Which we could certainly do, and we could certainly do it in a more efficient way than the current header parsing logic does (e.g. deal with spans instead of allocating strings). But I'm not sure the perf here matters much.

There could be (and almost certainly are) scenarios where this perf does matter, e.g. Connection: close. But again, it's not as simple as just using NonValidated.

In short: Yes, there are things we should look into as far as improving header access perf, but NonValidated by itself doesn't really help all that much, and we need to investigate further to determine if/how to actually do better here.

**Comment by geoffkizer** [comment]
> Is a test feasible?

Sadly, I cannot find a test for the existing behavior here. Given how small this change is, I think we should just take it despite the lack of tests.

Authentication tests in general are a pain because you need to have a certain machine setup to run them. But we do have some, and we should certainly add one that covers this case as well in the future.

@wfurt did I miss anything here? Thoughts?

File: src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs (hunk 1 above)
Comment (stephentoub) [reply src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs:64]: > Is the goal here to improve performance by avoiding the added processing/memory use that happens when you go through TryGetValues etc?

Yes, that would be the goal.

**Comment by ChrisFWood** [comment]
> 
> 
> Is a test feasible?

Fair question, I couldn't see anything testing that area to update, and I'm not that familiar with workings of authentication via proxy. In theory I could use reflection to test the private static method modified, rather than the whole authentication via proxy piece.
